cachi2/core/package_managers/bundler/main.py

```diff
@@ -99,3 +106,36 @@ def _get_repo_name_from_origin_remote(package_dir: RootedPath) -> str:
 
     resolved_path = Path(repo_name).joinpath(package_dir.subpath_from_root)
     return str(resolved_path)
+
+
+def _prepare_environment_variables_for_hermetic_build() -> list[EnvironmentVariable]:
+    return [
+        # Contains path to a directory where a new config could be found.
+        EnvironmentVariable(name="BUNDLE_APP_CONFIG", value="${output_dir}/" + CONFIG_OVERRIDE),
+    ]
+
+
+def _prepare_for_hermetic_build(source_dir: RootedPath, output_dir: RootedPath) -> None:
+    """Prepare a package for hermetic build by injecting necessary config."""
+    potential_bundle_config = source_dir.join_within_root(".bundle/config").path
```

Comment:
I have to come back again to my comment here https://github.com/containerbuildsystem/cachi2/pull/654#discussion_r1766353638.

If a user sets `ENV BUNDLE_APP_CONFIG=....` in his Dockerfile to some location other than `.bundler/config` we should check that, if he did, copy the settings and then overwrite the variable, right ?

Reply:
Are you suggesting we start parsing users' Dockerfiles? I don't think we want to get into that business given the complexity of multi-stage builds for this kind of validation. Besides, having `BUNDLE_APP_CONFIG` set in the Dockerfile makes no practical sense because you always want to be able to build your project in any environment and in a reproducible manner and allowing the build configuration to be changed like this goes against that principle (even without cachi2 in the picture), so I don't think we even need to concern ourselves with this scenario.

Reply:
No parsing. Just checking env variable. I know it is corner case, but everything feels like corner case here.

```
export BUNDLE_APP_CONFIG=foo
bundler config list
```

Reply:
> No parsing. Just checking env variable. I know it is corner case, but everything feels like corner case here.

How does this relate to the Dockerfile use case? What you've described is the local direct usage of cachi2 outside of a trusted pipeline containerized environment and my point

> Besides, having BUNDLE_APP_CONFIG set in the Dockerfile makes no practical sense because you always want to be able to build your project in any environment and in a reproducible manner and allowing the build configuration to be changed like this goes against that principle (even without cachi2 in the picture), so I don't think we even need to concern ourselves with this scenario.

applies to this scenario as well. AFAIK we don't do this for other pkg mgrs.

Reply:
I see it is already implemented.

Reply:
Yes, I have added this check. Admittedly this corner case looks somewhat contrived, as it would indicate a user doing something rather unusual, but the code to cover it is trivial and unlikely to introduce bugs so let's just have it.